src/sonic-bgpcfgd/tests/test_static_route.py

```diff
@@ -0,0 +1,307 @@
+    mgr = constructor()
+    set_del_test(
+        mgr,
+        "SET",
+        ("vrfRED|10.1.3.0/24", {
+            "nexthop": "10.0.0.57,10.0.0.59,10.0.0.61",
+            "ifname": "PortChannel0001,PortChannel0002,PortChannel0003",
+            "distance": "10,20,30",
+            "nexthop-vrf": "nh_vrf,,default",
+            "blackhole": "false,false,false",
+        }),
+        True,
+        [
+            "ip route 10.1.3.0/24 10.0.0.57 PortChannel0001 10 nexthop-vrf nh_vrf vrf vrfRED",
+            "ip route 10.1.3.0/24 10.0.0.59 PortChannel0002 20 vrf vrfRED",
+            "ip route 10.1.3.0/24 10.0.0.61 PortChannel0003 30 nexthop-vrf default vrf vrfRED"
+        ]
+    )
+    set_del_test(
+        mgr,
+        "SET",
+        ("vrfRED|10.1.3.0/24", {
+            "nexthop": "10.0.0.57,10.0.0.59,10.0.0.61",
+            "ifname": "PortChannel0001,PortChannel0002,PortChannel0003",
+            "distance": "40,50,60",
+            "nexthop-vrf": "nh_vrf,,default",
+            "blackhole": "false,false,false",
+        }),
+        True,
+        [
+            "no ip route 10.1.3.0/24 10.0.0.57 PortChannel0001 10 nexthop-vrf nh_vrf vrf vrfRED",
+            "no ip route 10.1.3.0/24 10.0.0.59 PortChannel0002 20 vrf vrfRED",
+            "no ip route 10.1.3.0/24 10.0.0.61 PortChannel0003 30 nexthop-vrf default vrf vrfRED",
+            "ip route 10.1.3.0/24 10.0.0.57 PortChannel0001 40 nexthop-vrf nh_vrf vrf vrfRED",
+            "ip route 10.1.3.0/24 10.0.0.59 PortChannel0002 50 vrf vrfRED",
+            "ip route 10.1.3.0/24 10.0.0.61 PortChannel0003 60 nexthop-vrf default vrf vrfRED"
+        ]
+    )
+
+@patch('bgpcfgd.manager_static_route.log_debug')
+def test_set_no_action(mocked_log_debug):
+    mgr = constructor()
+    set_del_test(
+        mgr,
+        "SET",
+        ("default|10.1.1.0/24", {
+            "nexthop": "10.0.0.57",
+            "ifname": "PortChannel0001",
+            "blackhole": "true",
+        }),
+        True,
+        [
+            "ip route 10.1.1.0/24 blackhole"
+        ]
+    )
+
+    set_del_test(
+        mgr,
+        "SET",
+        ("default|10.1.1.0/24", {
+            "nexthop": "10.0.0.59",
+            "ifname": "PortChannel0002",
+            "blackhole": "true",
+        }),
+        True,
+        []
+    )
+    mocked_log_debug.assert_called_with("Nothing to update for static route default|10.1.1.0/24")
+
+@patch('bgpcfgd.manager_static_route.log_debug')
+def test_del_no_action(mocked_log_debug):
+    mgr = constructor()
+    set_del_test(
+        mgr,
+        "DEL",
+        ("default|10.1.1.0/24",),
+        True,
+        []
+    )
+    mocked_log_debug.assert_called_with("Nothing to update for static route default|10.1.1.0/24")
+
+def test_set_invalid_arg():
+    mgr = constructor()
+    set_del_test(
+        mgr,
+        "SET",
+        ("default|10.1.1.0/24", {
+            "nexthop": "10.0.0.57, 10.0.0.59",
+            "ifname": "PortChannel0001",
```

Comment:
ifname is optional, so is nexthop.. so if either one is provided, you should consider it valid. But if user specifies both, then there should be empty string to specify positional arguments. Could you please add few cases for this as well?

Reply:
Yes, both ifname and nexthop are optional. In the case where both ifname and nexthop are given, they should have the number of items to avoid confusion. An empty string separated by a comma is needed if only one item has no value given, for example, `"nexthop": "10.0.0.57,10.0.0.59", "ifname": "PortChannel0001,"` is valid, `"nexthop": "10.0.0.57, 10.0.0.59", "ifname": "PortChannel0001"` is not. I added more test cases to better cover these scenarios.